javascript/javascript1/week3/main.js

```diff
@@ -0,0 +1,274 @@
+      ", has the following note text: " +
+      notes[i].content
+    );
+  }
+}
+
+console.log(logOutNotesFormatted());
+
+/*Unique feature: to remove a note once done 
+using content and returns newNotes with 
+remaining contents and id*/
+
+const newNotes = [...notes]; // make a clone of notes to remove content and to keep the original version of notes
+function removeNote(content) {
+  for (var i = 0; i < newNotes.length; i++) {
+    if (newNotes[i].content === content) {
+      newNotes.splice(i, 1);
+      i--;
+    }
+  }
+  return newNotes;
+}
+
+console.log(removeNote("Do laundry"));
+
+//CactusIO-interactive (Smart phone usage app) optional
+const activities = [];
+var i = 0;
+function addActivity(data, activity, duration) {
+  if (data === undefined || activity === undefined || duration === undefined) {
+    return "Add some activities before calling showStatus";
+  }
+  activities.push({});
+  for (; i < activities.length; i++) {
+    activities[i].data = data;
+    activities[i].activity = activity;
+    activities[i].duration = duration;
+  }
+}
+addActivity("23-7-2018", "Amazon Prime", 10);
+addActivity("23-7-2018", "Youtube", 30);
+addActivity("24-7-2018", "MTV", 30);
+addActivity("24-7-2018", "Youtube", 30);
+addActivity("26-7-2018", "Amazon Prime", 30);
+console.log(activities);
+
+var totalUsage = 0;
+var j = 0; // extra feature to calculate with how many activities the usageLimite is reached.
+const usageLimit = 120;
+function showStatus() {
+  const totalActivities = activities.length;
+  for (var i = 0; i < activities.length; i++) {
+    totalUsage += activities[i].duration;
+    if (j === 0) {
+      if (totalUsage > usageLimit || totalUsage === usageLimit) {
+        var limitOfActivity = i + 1;
+        console.log(
+          "You have reached your usage limit with first " +
+            limitOfActivity +
+            " activities of your activities list, no more smartphoning for you!"
+        );
+        j = j + 1;
+      }
+    }
+  }
+  return [totalActivities, totalUsage];
+}
+const showMyStatus = showStatus(activities);
+console.log(
+  "You have added " +
+    showMyStatus[0] +
+    " activities. They amount to " +
+    showMyStatus[1] +
+    " min. of usage"
+);
+//FOR OPTIONAL FEATURES:
+const newActivities = [];
+var i = 0;
+var today = new Date();
+var todaysDate =
+  today.getDate() + "-" + (today.getMonth() + 1) + "-" + today.getFullYear();
+function addActivityWithoutDate(activity, duration) {
+  if (activity === undefined || duration === undefined) {
+    return "Add some activities before calling showStatus";
+  }
+  newActivities.push({});
+  for (; i < newActivities.length; i++) {
+    newActivities[i].date = todaysDate;
+    newActivities[i].activity = activity;
+    newActivities[i].duration = duration;
+  }
+}
+addActivityWithoutDate("Youtube", 30);
+addActivityWithoutDate("Facebook", 40);
+addActivityWithoutDate("News", 30);
+
+console.log(newActivities);
+
+var k = 0;
+function newShowStatus() {
```

Comment:
The previous showStatus showed the activites themselves, not just the number of them. I expected this one to do the same. You could make a new array and add the activities that pass the `newActivities[i].date === todaysDate` check to the array and then return the array

Reply:
In the home work, it was mentioned: "Improve the showStatus function by only showing the number of activities for that specific day." But now, I made a new array and add the activities that pass the newActivities[i].date === todaysDate check and logged it out outside.